nifi-stateless/nifi-stateless-bundle/nifi-stateless-engine/src/main/java/org/apache/nifi/stateless/queue/StatelessFlowFileQueue.java

```diff
@@ -283,8 +283,8 @@ public String getFlowFileExpiration() {
     }
 
     @Override
-    public int getFlowFileExpiration(final TimeUnit timeUnit) {
-        return (int) timeUnit.convert(expirationMillis, TimeUnit.MILLISECONDS);
+    public long getFlowFileExpiration(final TimeUnit timeUnit) {
+        return timeUnit.convert(expirationMillis, TimeUnit.MILLISECONDS);
```

Comment:
I believe a change also needs to be made to   _public void setFlowFileExpiration(final String flowExpirationPeriod)_, as the value on line 292 is still being cast to int.